### DISCUSSION TITLE
How to test action using github.head_ref

### DISCUSSION THREAD

**Discussion by andy5995** [community/community#35812] [Actions] [answered]
When I'm making changes to an action I'm working on and open a PR, I want it to test the action using the head_ref, e.g.

`uses: ...action@${{ github.head_ref }}```

But that results in an error when the runner attempts to start it:

>The workflow is not valid. .github/workflows/test.yml (Line: 30, Col: 13): Unrecognized named-value: 'github'. Located at position 1 within expression: github.head_ref .github/workflows/test.yml (Line: 45, Col: 13): Unrecognized named-value: 'github'. Located at position 1 within expression: github.head_ref

**Comment by airtower-luna** [comment, marked as answer]
You can't set action versions dynamically. Maybe set up the test so the action [runs as a local action](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow), as it's present in the checked out repository?
